cirq rigetti cause CI failure for pytest and mypy at head

pytest check: https://github.com/quantumlib/Cirq/actions/runs/10922443135/job/30316787499?pr=6728
```
FAILED cirq-core/cirq/protocols/json_serialization_test.py::test_json_and_repr_data[mod_spec45-/home/runner/work/Cirq/Cirq/cirq-rigetti/cirq_rigetti/json_test_data/RigettiQCSAspenDevice] - OSError: Failed to parse test json data from cirq-rigetti/cirq_rigetti/json_test_data/RigettiQCSAspenDevice.json.
FAILED cirq-core/cirq/protocols/json_serialization_test.py::test_json_test_data_coverage[mod_spec270-RigettiQCSAspenDevice-RigettiQCSAspenDevice] - AttributeError: type object 'builtins.Family' has no attribute 'Aspen'
```


type check: https://github.com/quantumlib/Cirq/actions/runs/10922443135/job/30316786280?pr=6728

```
cirq-rigetti/cirq_rigetti/aspen_device.py:69: error: "type[Family]" has no attribute "Aspen"  [attr-defined]
cirq-rigetti/cirq_rigetti/aspen_device_test.py:256: error: "type[Family]" has no attribute "NONE"  [attr-defined]
cirq-rigetti/cirq_rigetti/aspen_device_test.py:259: error: "type[Family]" has no attribute "Aspen"  [attr-defined]
```

---

blocking https://github.com/quantumlib/Cirq/pull/6728

--- a/cirq-rigetti/cirq_rigetti/aspen_device.py
+++ b/cirq-rigetti/cirq_rigetti/aspen_device.py
@@ -66,7 +66,7 @@ class RigettiQCSAspenDevice(cirq.devices.Device):
         else:
             self.isa = InstructionSetArchitecture.from_raw(json.dumps(isa))
 
-        if self.isa.architecture.family != Family.Aspen:
+        if not Family.is_aspen(self.isa.architecture.family):
             raise UnsupportedRigettiQCSQuantumProcessor(
                 'this integration currently only supports Aspen devices, '
                 f'but client provided a {self.isa.architecture.family} device'

--- a/cirq-rigetti/cirq_rigetti/aspen_device_test.py
+++ b/cirq-rigetti/cirq_rigetti/aspen_device_test.py
@@ -253,10 +253,10 @@ def test_rigetti_qcs_aspen_device_repr(qcs_aspen8_isa: InstructionSetArchitectur
 def test_rigetti_qcs_aspen_device_family_validation(qcs_aspen8_isa: InstructionSetArchitecture):
     """test RigettiQCSAspenDevice validates architecture family on initialization"""
     non_aspen_isa = InstructionSetArchitecture.from_raw(qcs_aspen8_isa.json())
-    non_aspen_isa.architecture.family = Family.NONE
+    non_aspen_isa.architecture.family = Family.new_none()
 
-    assert (
-        non_aspen_isa.architecture.family == Family.Aspen
+    assert Family.is_aspen(
+        non_aspen_isa.architecture.family
     ), 'ISA family is read-only and should still be Aspen'
 
 

--- a/cirq-rigetti/requirements.txt
+++ b/cirq-rigetti/requirements.txt
@@ -1,1 +1,4 @@
 pyquil>=4.11.0,<5.0.0
+
+# TODO - remove once pyquil requires qcs-sdk-python >= 0.20.1
+qcs-sdk-python>=0.20.1

